vllm/v1/cudagraph_dispatcher.py

```diff
@@ -94,12 +117,38 @@ def initialize_cudagraph_keys(
                 if x <= max_num_tokens and x >= uniform_decode_query_len
             ]
             for bs in cudagraph_capture_sizes_for_decode:
+                num_reqs = self.calculate_num_reqs_for_tokens(
+                    bs, uniform_decode_query_len, True
+                )
                 self.add_cudagraph_key(
                     CUDAGraphMode.FULL,
-                    BatchDescriptor(num_tokens=bs, uniform_decode=True),
+                    BatchDescriptor(
+                        num_tokens=bs, uniform_decode=True, num_reqs=num_reqs
+                    ),
                 )
+
         self.keys_initialized = True
 
+    def calculate_num_reqs_for_tokens(
```

Comment:
I don't think I understand what this does. It gives the maximum possible number of non-empty requests? But I thought we also have requests of length 0 sometimes?